extension/src/XliffFunctions.ts

```diff
@@ -299,68 +295,62 @@ export function refreshSelectedXlfFileFromGXlf(
           }
           refreshResult.numberOfUpdatedNotes++;
         }
-        if (langTransUnit.sourceIsEmpty() && langTransUnit.targetIsEmpty()) {
+        if (
+          langTransUnit.sourceIsEmpty() &&
+          langTransUnit.targetIsEmpty() &&
+          lfSettings.preferLockedTranslations
+        ) {
           langTransUnit.insertCustomNote(
             CustomNoteType.refreshXlfHint,
             RefreshXlfHint.emptySource
           );
-          refreshResult.numberOfReviewsAdded++;
+          langTransUnit.target.translationToken = TranslationToken.review;
         }
         formatTransUnitForTranslationMode(
-          languageFunctionsSettings.translationMode,
+          lfSettings.translationMode,
           langTransUnit
         );
-        detectInvalidValues(langTransUnit, languageFunctionsSettings);
+        detectInvalidValues(langTransUnit, lfSettings);
         if (langTransUnit.needsReview(true)) {
           refreshResult.numberOfReviewsAdded++;
         }
       }
       newLangXliff.transunit.push(langTransUnit);
       langXliff.transunit.splice(langXliff.transunit.indexOf(langTransUnit), 1); // Remove all handled TransUnits -> The rest will be deleted.
-    } else {
+    } else if (!sortOnly) {
       // Does not exist in target
-      if (!sortOnly) {
-        const newTransUnit = TransUnit.fromString(gTransUnit.toString());
-        newTransUnit.targets = [];
-        newTransUnit.targets.push(
-          getNewTarget(
-            languageFunctionsSettings.translationMode,
-            langIsSameAsGXlf,
-            gTransUnit
-          )
-        );
-        if (langIsSameAsGXlf) {
-          newTransUnit.insertCustomNote(
-            CustomNoteType.refreshXlfHint,
-            RefreshXlfHint.newCopiedSource
-          );
-        } else {
-          newTransUnit.insertCustomNote(
-            CustomNoteType.refreshXlfHint,
-            RefreshXlfHint.new
-          );
-        }
-        if (newTransUnit.sourceIsEmpty()) {
-          newTransUnit.insertCustomNote(
-            CustomNoteType.refreshXlfHint,
-            RefreshXlfHint.emptySource
-          );
-        }
-        formatTransUnitForTranslationMode(
-          languageFunctionsSettings.translationMode,
-          newTransUnit
+      const newTransUnit = TransUnit.fromString(gTransUnit.toString());
+      newTransUnit.targets = [];
+      newTransUnit.targets.push(
+        getNewTarget(lfSettings.translationMode, langIsSameAsGXlf, gTransUnit)
+      );
+      const hintText = langIsSameAsGXlf
+        ? RefreshXlfHint.newCopiedSource
+        : RefreshXlfHint.new;
+      newTransUnit.insertCustomNote(CustomNoteType.refreshXlfHint, hintText);
+      newTransUnit.target.translationToken = TranslationToken.notTranslated;
```

Comment:
Is this really needed? I mean, how could it work before then? 
The state should already be set in `getNewTarget()`, or do we have an issue there?

Reply:
I think we do. I'll need to retrace my steps to reproduce the test I used. I'll get back to you on this.

Reply:
You were correct. I don't know how I came to that conclusion but it was not correct :)